ksql-streams/src/main/java/io/confluent/ksql/execution/streams/StreamStreamJoinBuilder.java

```diff
@@ -46,7 +46,7 @@ private StreamStreamJoinBuilder() {
     final QueryContext.Stacker stacker = QueryContext.Stacker.of(queryContext);
     final LogicalSchema leftSchema = left.getSchema();
     final PhysicalSchema leftPhysicalSchema = PhysicalSchema.from(
-        leftSchema.withoutAlias(),
+        leftSchema,
```

Comment:
This will likely cause a performance regression for json schemas, since the full source name will be written into the json field names (rather than just the column names). However I think this is better optimized in the json serde by dropping any prefix common to all field names, or by using our own internally generated field names for internal topics.